plugins/modules/idrac_license.py

```diff
@@ -0,0 +1,824 @@
+            job_details=job_dict)
+    return job_dict
+
+
+def get_share_details(module):
+    share_details = {}
+    share_details["IPAddress"] = module.params.get('share_parameters').get('ip_address')
+    share_details["ShareName"] = module.params.get('share_parameters').get('share_name')
+    share_details["UserName"] = module.params.get('share_parameters').get('username')
+    share_details["Password"] = module.params.get('share_parameters').get('password')
+    return share_details
+
+
+def get_proxy_details(module):
+    proxy_details = {}
+    if module.params.get('share_parameters').get('share_type') == "http":
+        proxy_details["ShareType"] = "HTTP"
+    else:
+        proxy_details["ShareType"] = "HTTPS"
+    proxy_details["IPAddress"] = module.params.get('share_parameters').get('ip_address')
+    proxy_details["ShareName"] = module.params.get('share_parameters').get('share_name')
+    proxy_details["UserName"] = module.params.get('share_parameters').get('username')
+    proxy_details["Password"] = module.params.get('share_parameters').get('password')
+    proxy_details["IgnoreCertWarning"] = IGNORE_CERTIFICATE_WARNING[module.params.get('share_parameters').get('ignore_certificate_warning')]
+    if module.params.get('share_parameters').get('proxy_support') == "parameters_proxy":
+        proxy_details["ProxySupport"] = PROXY_SUPPORT[module.params.get('share_parameters').get('proxy_support')]
+        proxy_details["ProxyType"] = PROXY_TYPE[module.params.get('share_parameters').get('proxy_type')]
+        proxy_details["ProxyServer"] = module.params.get('share_parameters').get('proxy_server')
+        proxy_details["ProxyPort"] = module.params.get('share_parameters').get('proxy_port')
+        if module.params.get('share_parameters').get('proxy_username') and module.params.get('share_parameters').get('proxy_password'):
+            proxy_details["ProxyUname"] = module.params.get('share_parameters').get('proxy_username')
+            proxy_details["ProxyPasswd"] = module.params.get('share_parameters').get('proxy_password')
+    return proxy_details
+
+
+def main():
+    specs = get_argument_spec()
+    specs.update(idrac_auth_params)
+    module = AnsibleModule(
+        argument_spec=specs,
+        mutually_exclusive=[("import", "export", "delete")],
+        required_if=[
+            ["import", True, ("share_parameters",)],
+            ["export", True, ("license_id", "share_parameters",)],
+            ["delete", True, ("license_id",)]
+        ],
+        supports_check_mode=False
+    )
+
+    try:
+        with iDRACRedfishAPI(module.params) as idrac:
+            idrac_firmware_version = get_idrac_firmware_version(idrac)
+            if LooseVersion(idrac_firmware_version) <= '3.0':
+                module.exit_json(msg="iDRAC firmware version is not supported.", failed=True)
+            license_obj = None
+            if module.params["import"]:
+                license_obj = ImportLicense(idrac, module)
```

Comment:
Reduce the if else statement and call the class/function dynamically in runtime using globals() and locals() which I told you yesterday, check other places too. We can reduce so many line of code.

Reply:
Done